### DISCUSSION TITLE
Any tips on how to update to v7 from v4.10.0?

### DISCUSSION THREAD

**Discussion by wyspar** [TypedProject/ts-express-decorators#2345] [Q&A] [answered]
I need to update the tsed packages because the json-schema 0.2.5 has prototype pollution. I noticed the latest version of v4 and v5 still use this version of json-schema 0.2.5

I know of some things already since I have started this process. Like importing Service comes from "@tsed/di" now.
ParamRegistry.decorate() is now deprecated from v4 to v5.

**Comment by Romakita** [comment]
Hello @wyspar 

the json-schema is just used as type declaration. The code isn’t executed by tsed.

there is no a strict equivalent to ParamRegistry.decorate and honestly I don’t remember what is the exact behavior of this function.

See you

**Reply by wyspar** [reply]
Thank you for the response. I tried to updating json-schema by using --save and putting it under overrides:{} but Veracode does not approve it that. Since this old version is a child dependency of tsed v4.10.0, so I think I have no choice but to update. 
Also I am using ParamRegistry.decorate() like this:

ˋ ˋ ˋts
function myFunction(expression: string, useType:string ): Function {
 return ParamRegistry.decorate(userFilter, {expression, useType, useConverter: true, useValidation: true});
}

ˋ ˋ ˋ

then we call this function in post methods in the controllers. 

I can't post anymore code than this I think sadly.

**Reply by Romakita** [reply, marked as answer]
Hum ok the Filter is now called Pipe in v7:
https://tsed.io/docs/pipes.html
you have to rewrite your Filter to a Pipe using Pipe API. It should works, because Pipe is more flexible rather than the legacy Filter. ;)

See you
Romain
